src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -159,24 +160,32 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
         rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
       })
 
+      if (bn(withdrawBps).isZero()) return
+
       const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
 
       const { dust_amount, expected_amount_out } = quote
-      const withdrawFee = amountCryptoThorBaseUnit
-        .minus(expected_amount_out)
-        .div(bn(10).pow(THOR_PRECISION))
-
-      const dustAmountCryptoPrecision = fromThorBaseUnit(dust_amount)
 
-      setWithdrawFee(withdrawFee.toFixed())
-      setDustAmount(dustAmountCryptoPrecision.toFixed(asset.precision))
+      setWithdrawFeeCryptoBaseUnit(
+        toBaseUnit(
+          fromThorBaseUnit(amountCryptoThorBaseUnit.minus(expected_amount_out)),
+          asset.precision,
+        ),
+      )
+      setDustAmountCryptoBaseUnit(
```

Comment:
both @kaladinlight and i are wondering if this is going to cause infinite renders?

would a better approach be a) memoize the quote, and b) memoize these `dustAmountCryptoBaseUnit` and `withdrawFeeCryptoBaseUnit` as separate values, rather than storing in state?

Reply:
nvm can't do async useMemo and these are primitives anyway 👍